lib/test_prof/recipes/rspec/let_it_be.rb

```diff
@@ -145,7 +210,35 @@ def define_let_it_be_methods(identifier, **modifiers)
       end
       record
     end
+
+    config.register_modifier :freeze do |record, val|
+      # TODO: change this to `if val == false` when TestProf hits 1.0
+      unless val == true
+        TestProf::LetItBe::Stoplist.push(record)
+        next record
+      end
+
+      TestProf::LetItBe::Freezer.deep_freeze(record)
+      record
+    end
   end
 end
 
 RSpec::Core::ExampleGroup.extend TestProf::LetItBe
+RSpec.configure do |config|
+  config.after(:example) do |example|
+    if example.exception&.message&.match?(/can't modify frozen/)
+      example.exception.message << TestProf::LetItBe::FROZEN_ERROR_HINT
+    end
+  end
+end
+
+TestProf::BeforeAll.configure do |config|
+  config.before(:begin) do
+    TestProf::LetItBe::Stoplist.begin
+  end
+
+  config.after(:rollback) do
+    TestProf::LetItBe::Stoplist.rollback
+  end
```

Comment:
I couldn't find a hook that would run on the outermost `begin` (where the first `before_all`'s block is called), or have a context to call `within_before_all?` for the sake of resetting the stoplist.
That made me rework the stoplist as a layer per example group structure, and use those two hooks to add and remove layers.